doc/src/lexer_tutorial/002_raw_delimited_content.md

```diff
@@ -0,0 +1,347 @@
+Consider this variant of the previous test:
+
+```rust
+#[test]
+fn foul_ball() {
+    assert_eq!(nobol2::StmtParser::new().parse(r#"z = "x""#), Ok(('z', "x").into()));
+}
+```
+
+The above produces:
+
+```
+---- foul_ball stdout ----
+thread 'foul_ball' panicked at 'assertion failed: `(left == right)`
+  left: `Err(UnrecognizedToken { token: (5, Token(3, "x"), 6), expected: ["r#\"[a-z]*\"#"] })`,
+ right: `Ok(Stmt(Var('z'), Lit("x")))`', doc/nobol/src/main.rs:43:5
+```
+
+What is the problem?
+
+Merely specifying a precedence to favor tokenizing `r"[x-z]"` over `r"[a-z]*"`
+does not address the real problem here. That precedence rule causes an input
+like `z = "x"` to be split into tokens such that the `x` only matches the
+regular expression for the `Var`. It will not match the `r"[a-z]*"` in the `Lit`
+rule, even if it intuitively seems like it should; they have already been
+lexically categorized as different tokens at this point.
+
+One could add further workarounds to deal with this. For example, one could
+change the `Lit` production to explicitly handle the `r"[x-z]"` regular
+expression as its own case:
+
+```lalrpop
+pub Lit: Lit = {
+    "\"" <r"[x-z]"> "\"" => <>.into(),
+    "\"" <r"[a-z]*"> "\"" => <>.into(),
+};
+```
+
+But this is a fragile workaround.
+
+Specifically, this workaround is only applicable because we put artificial
+limits in this language.
+
+If we wanted to generalize string literals to be able to contain other
+characters (such as whitespace), the technique described so far does not work
+out well. Consider this grammar:
+
+```lalrpop
+match {
+   r"[x-z]"
+} else {
+   r"[a-z ]*",
+   _
+}
+
+pub Var: Var = <r"[x-z]"> => <>.chars().next().unwrap().into();
+
+pub Lit: Lit = {
+    "\"" <r"[x-z]"> "\"" => <>.into(),
+    "\"" <r"[a-z ]*"> "\"" => <>.into(),
+};
+
+pub Stmt: Stmt = <Var> "=" <Lit> => (<>).into();
+```
+
+Now, if we run the same test as before:
+
+```rust
+#[test]
+fn spaceballs() {
+    assert_eq!(nobol4::StmtParser::new().parse(r#"z = "x""#), Ok(('z', "x").into()));
+}
+```
+
+we get the following error output:
+
+```
+thread 'spaceballs' panicked at 'assertion failed: `(left == right)`
+  left: `Err(UnrecognizedToken { token: (0, Token(2, "z "), 2), expected: ["r#\"[x-z]*\"#"] })`,
+ right: `Ok(Stmt(Var('z'), Lit("x")))`', doc/nobol/src/main.rs:58:5
+```
+
+Our attempt to generalize what string contents can be has caused problems for
+how the *rest* of the input is tokenized.
+
+
+#### The right way to do this
+
+Let us revisit the original rule in the grammar for string literals, from our
+first version:
+
+```lalrpop
+pub Lit: Lit = "\"" <r"[a-c]*"> "\"" => <>.into();
+```
+
+The heart of our problem is that we have implicitly specified distinct tokens
+for the string delimiter (`"\""`) versus the string content (in this case,
+`r"[a-c]*"`).
+
+The problem is that our intuition is that we only want to tokenize string
```

Comment:
```suggestion
Our intuition is that we only want to tokenize string
```

Reply:
Addressed in fb568a3be286438944569aae7d25b33e6e0fa17b